### ISSUE TITLE
Deserialization of optional types is still required by serde-yaml

### ISSUE THREAD

**Issue by calebmer** [dtolnay/serde-yaml#8]
Take the following example:

```rust
#[derive(Deserialize)]
struct Data {
  a: Option<f64>,
  b: Option<String>,
  c: bool
}
```

and the following YAML:

```yaml
b:
c: true
```

For this I would expect an object initialized like the following:

```rust
Data {
  a: None,
  b: Some("".to_string()),
  c: true
}
```

Instead `serde_yaml` throws an error that `a` is not defined.

**Comment by dtolnay** [comment]
This is a feature of [serde](https://github.com/serde-rs/serde) (not serde-yaml) that enforces that all fields exist when deserializing. If you want `a` to be set to its default value when not specified, you need to annotate it with `#[serde(default)]`. See https://github.com/serde-rs/serde#annotations

This does what you want:

```rust
#[derive(Deserialize)]
struct Data {
    #[serde(default)]
    a: Option<f64>,
    b: Option<String>,
    c: bool
}
```

**Comment by calebmer** [comment]
`serde-json` behaves this way for `Some`/`None`, that's why I filed the issue here. This seems like a common enough use case that it should be accommodated. Should I open an issue on `serde`?

**Comment by dtolnay** [comment]
Thanks - I didn't realize serde-json handles this. I will look into how they do it.

**Comment by dtolnay** [comment]
@calebmer I have handled this case the same way that serde-json does. I have released [0.2.2](https://crates.io/crates/serde_yaml/0.2.2) which includes the fix.

**Comment by calebmer** [comment]
Thank you 😀
